tarofreighter/chain_porter.go

```diff
@@ -650,27 +652,47 @@ func (p *ChainPorter) stateStep(currentPkg sendPackage) (*sendPackage, error) {
 		if err != nil {
 			return nil, err
 		}
-		senderScriptKey, err := p.cfg.KeyRing.DeriveNextKey(
-			ctx, tarogarden.TaroKeyFamily,
-		)
-		if err != nil {
-			return nil, err
-		}
 
-		// We'll assume BIP 86 everywhere, and use the tweaked key from
-		// here on out.
-		currentPkg.SenderScriptKey = asset.NewScriptKeyBIP0086(
-			senderScriptKey,
-		)
+		// If we are sending the full value of the input asset, we will
+		// need to create a split with unspendable change.
+		if inputAsset.Type == asset.Normal && !needsSplit {
+			emptyRootKey, err := btcec.ParsePubKey(
+				asset.NScriptKey[:],
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			currentPkg.SenderScriptKey = asset.ScriptKey{
```

Comment:
👍 

I think having zeroes here for the key family and index are ok, as those are only used when we want to generate a sig, and we can't generate one for this type of key.